Create CONTRIBUTING.md (#187)
fedora-infra/mote: CONTRIBUTING.md

> And if you like the project, but just don't have time to contribute, that's fine. There are other easy ways to support the project and show your appreciation, which we would also be very happy about:
> - Star the project
> - Tweet about it
> - Refer this project in your project's readme
> - Mention the project at local meetups and tell your friends/colleagues

<!-- omit in toc -->
## Table of Contents

- [I Have a Question](#i-have-a-question)
- [I Want To Contribute](#i-want-to-contribute)
- [Reporting Bugs](#reporting-bugs)
- [Suggesting Enhancements](#suggesting-enhancements)
- [Your First Code Contribution](#your-first-code-contribution)
- [Improving The Documentation](#improving-the-documentation)
- [Styleguides](#styleguides)
- [Commit Messages](#commit-messages)
- [Join The Project Team](#join-the-project-team)

## I Have a Question

> If you want to ask a question, we assume that you have read the available [Documentation](https://github.com/fedora-infra/mote/blob/main/CODE_OF_CONDUCT.md).

> Before you ask a question, it is best to search for existing [Issues](https://github.com/fedora-infra/moteissues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue. It is also advisable to search the internet for answers first.

Reviewer: ```suggestion
Before you ask a question, it is best to search for existing [Issues](https://github.com/fedora-infra/mote/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue. It is also advisable to search the internet for answers first.
```

If you then still feel the need to ask a question and need clarification, we recommend the following:

> - Open an [Issue](https://github.com/fedora-infra/moteissues/new).

Reviewer: ```suggestion
- Open an [Issue](https://github.com/fedora-infra/mote/issues/new).
```

- Provide as much context as you can about what you're running into.

> - Provide project and platform versions (nodejs, npm, etc), depending on what seems relevant.

Reviewer: ```suggestion
- Provide project and platform versions (Python, PIP etc), depending on what seems relevant.
```

We will then take care of the issue as soon as possible.

> <!--
> You might want to create a separate issue tag for questions and include it in this description. People should then tag their issues accordingly.
> 
> Depending on how large the project is, you may want to outsource the questioning, e.g. to Stack Overflow or Gitter. You may add additional contact and information possibilities:
> - IRC
> - Slack
> - Gitter
> - Stack Overflow tag
> - Blog
> - FAQ
> - Roadmap
> - E-Mail List
> - Forum
> -->

Reviewer: ```suggestion
```

## I Want To Contribute

> ### Legal Notice <!-- omit in toc -->
> When contributing to this project, you must agree that you have authored 100% of the content, that you have the necessary rights to the content and that the content you contribute may be provided under the project license.

### Reporting Bugs

> <!-- omit in toc -->

Reviewer: ```suggestion
```

#### Before Submitting a Bug Report

A good bug report shouldn't leave others needing to chase you up for more information. Therefore, we ask you to investigate carefully, collect information and describe the issue in detail in your report. Please complete the following steps in advance to help us fix any potential bug as fast as possible.

- Make sure that you are using the latest version.

> - Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [documentation](https://github.com/fedora-infra/mote/blob/main/CODE_OF_CONDUCT.md). If you are looking for support, you might want to check [this section](#i-have-a-question)).

Reviewer: ```suggestion
- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [documentation](https://github.com/fedora-infra/mote/blob/main/README.md). If you are looking for support, you might want to check [this section](#i-have-a-question)).
```

> - To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](https://github.com/fedora-infra/moteissues?q=label%3Abug).

Reviewer: ```suggestion
- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](https://github.com/fedora-infra/mote/issues?q=label%3Abug).
```

- Also make sure to search the internet (including Stack Overflow) to see if users outside of the GitHub community have discussed the issue.
- Collect information about the bug:
- Stack trace (Traceback)
- OS, Platform and Version (Windows, Linux, macOS, x86, ARM)

> - Version of the interpreter, compiler, SDK, runtime environment, package manager, depending on what seems relevant.

Reviewer: ```suggestion
```

- Possibly your input and the output
- Can you reliably reproduce the issue? And can you also reproduce it with older versions?

<!-- omit in toc -->
#### How Do I Submit a Good Bug Report?

> You must never report security related issues, vulnerabilities or bugs to the issue tracker, or elsewhere in public. Instead sensitive bugs must be sent by email to <>.

> <!-- You may add a PGP key to allow the messages to be sent encrypted as well. -->

Reviewer: ```suggestion
```

We use GitHub issues to track bugs and errors. If you run into an issue with the project:

> - Open an [Issue](https://github.com/fedora-infra/moteissues/new). (Since we can't be sure at this point whether it is a bug or not, we ask you not to talk about a bug yet and not to label the issue.)

Reviewer: ```suggestion
- Open an [Issue](https://github.com/fedora-infra/mote/issues/new). (Since we can't be sure at this point whether it is a bug or not, we ask you not to talk about a bug yet and not to label the issue.)
```

- A team member will try to reproduce the issue with your provided steps. If there are no reproduction steps or no obvious way to reproduce the issue, the team will ask you for those steps and mark the issue as `needs-repro`. Bugs with the `needs-repro` tag will not be addressed until they are reproduced.
- If the team is able to reproduce the issue, it will be marked `needs-fix`, as well as possibly other tags (such as `critical`), and the issue will be left to be [implemented by someone](#your-first-code-contribution).

<!-- You might want to create an issue template for bugs and errors that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->

### Suggesting Enhancements

This section guides you through submitting an enhancement suggestion for mote, **including completely new features and minor improvements to existing functionality**. Following these guidelines will help maintainers and the community to understand your suggestion and find related suggestions.

<!-- omit in toc -->
#### Before Submitting an Enhancement

- Make sure that you are using the latest version.

> - Read the [documentation](https://github.com/fedora-infra/mote/blob/main/CODE_OF_CONDUCT.md) carefully and find out if the functionality is already covered, maybe by an individual configuration.

Reviewer: ```suggestion
- Read the [documentation](https://github.com/fedora-infra/mote/blob/main/README.md) carefully and find out if the functionality is already covered, maybe by an individual configuration.
```

> - Perform a [search](https://github.com/fedora-infra/moteissues) to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.

Reviewer: ```suggestion
- Perform a [search](https://github.com/fedora-infra/mote/issues) to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.
```

- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to convince the project's developers of the merits of this feature. Keep in mind that we want features that will be useful to the majority of our users and not just a small subset. If you're just targeting a minority of users, consider writing an add-on/plugin library.

> <!-- omit in toc -->

Reviewer: ```suggestion
```

#### How Do I Submit a Good Enhancement Suggestion?

> Enhancement suggestions are tracked as [GitHub issues](https://github.com/fedora-infra/moteissues).

Reviewer: ```suggestion
Enhancement suggestions are tracked as [GitHub issues](https://github.com/fedora-infra/mote/issues).
```

- Use a **clear and descriptive title** for the issue to identify the suggestion.
- Provide a **step-by-step description of the suggested enhancement** in as many details as possible.
- **Describe the current behavior** and **explain which behavior you expected to see instead** and why. At this point you can also tell which alternatives do not work for you.
- You may want to **include screenshots and animated GIFs** which help you demonstrate the steps or point out the part which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux. <!-- this should only be included if the project has a GUI -->
- **Explain why this enhancement would be useful** to most mote users. You may also want to point out the other projects that solved it better and which could serve as inspiration.

> <!-- You might want to create an issue template for enhancement suggestions that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->

Reviewer: ```suggestion
```

> <!-- You might want to create an issue template for enhancement suggestions that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->

Reviewer: ```suggestion
```

> ### Your First Code Contribution
> <!-- TODO
> include Setup of env, IDE and typical getting started instructions?
> 
> -->
> 
> ### Improving The Documentation
> <!-- TODO
> Updating, improving and correcting the documentation
> 
> -->
> 
> ## Styleguides
> ### Commit Messages
> <!-- TODO
> 
> -->
> 
> ## Join The Project Team
> <!-- TODO -->

Reviewer: ```suggestion
```
